File: routestory/src/main/java/net/routestory/parts/HapticButton.java
package net.routestory.parts;

import android.content.Context;
import android.util.AttributeSet;
import android.view.HapticFeedbackConstants;
import android.view.View;
import android.widget.Button;

public class HapticButton extends Button {
	public HapticButton(Context context) {
		super(context);
	}
	
	public HapticButton (Context context, AttributeSet attrs) {
	    super(context, attrs);
	}

	public HapticButton (Context context, AttributeSet attrs, int defStyle) {
	    super(context, attrs, defStyle);
	}
	
	@Override
	public void setOnClickListener(final View.OnClickListener l) {
		super.setOnClickListener(new View.OnClickListener() {
			public void onClick(View v) {
				v.setHapticFeedbackEnabled(true);
				v.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY);
				l.onClick(v);
			}
		});
	}
}

File: src/main/scala/net/routestory/ui/HapticButton.scala
package net.routestory.ui

import android.content.Context
import android.widget.Button
import android.view.View.OnClickListener
import android.view.{ HapticFeedbackConstants, View }

class HapticButton(ctx: Context) extends Button(ctx) {
  override def setOnClickListener(l: OnClickListener) {
    super.setOnClickListener(new OnClickListener {
      def onClick(v: View) {
        v.setHapticFeedbackEnabled(true)
        v.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY)
        l.onClick(v)
      }
    })
  }
}
